lib/modules/platform/gerrit/index.ts

```diff
@@ -0,0 +1,544 @@
+import JSON5 from 'json5';
+import type { RepoGlobalConfig } from '../../../config/types';
+import { logger } from '../../../logger';
+import type { BranchStatus } from '../../../types';
+import * as git from '../../../util/git';
+import { setBaseUrl } from '../../../util/http/gerrit';
+import { regEx } from '../../../util/regex';
+import { ensureTrailingSlash } from '../../../util/url';
+import type {
+  BranchStatusConfig,
+  CreatePRConfig,
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfigByContent,
+  EnsureCommentRemovalConfigByTopic,
+  EnsureIssueConfig,
+  EnsureIssueResult,
+  FindPRConfig,
+  Issue,
+  MergePRConfig,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { repoFingerprint } from '../util';
+
+import { smartTruncate } from '../utils/pr-body';
+import readOnlyIssueBody from '../utils/read-only-issue-body';
+import { client } from './client';
+import { configureScm } from './scm';
+import type {
+  GerritChange,
+  GerritFindPRConfig,
+  GerritLabelTypeInfo,
+  GerritProjectInfo,
+} from './types';
+import {
+  TAG_PULL_REQUEST_BODY,
+  getGerritRepoUrl,
+  mapBranchStatusToLabel,
+  mapGerritChangeToPr,
+  mapPrStateToGerritFilter,
+} from './utils';
+
+export const id = 'gerrit';
+
+const defaults: {
+  endpoint?: string;
+} = {};
+
+let config: {
+  repository?: string;
+  head?: string;
+  config?: GerritProjectInfo;
+  labels: Record<string, GerritLabelTypeInfo>;
+  labelMappings?: {
+    stabilityDaysLabel?: string;
+    mergeConfidenceLabel?: string;
+  };
+  gerritUsername?: string;
+} = {
+  labels: {},
+  labelMappings: {
+    stabilityDaysLabel: 'Renovate-Stability',
+  },
+};
+
+export function mergeToConfig(newConfig: typeof config): void {
+  config = { ...config, ...newConfig };
+}
+
+export function initPlatform({
+  endpoint,
+  username,
+  password,
+  gerritLabelMapping,
+}: PlatformParams & RepoGlobalConfig): Promise<PlatformResult> {
+  logger.debug(`initPlatform(${endpoint!}, ${username!})`);
+  if (!endpoint) {
+    throw new Error('Init: You must configure a Gerrit Server endpoint');
+  }
+  if (!(username && password)) {
+    throw new Error(
+      'Init: You must configure a Gerrit Server username/password'
+    );
+  }
+  config.labelMappings = gerritLabelMapping;
+  config.gerritUsername = username;
+  defaults.endpoint = ensureTrailingSlash(endpoint);
+  setBaseUrl(defaults.endpoint);
+  const platformConfig: PlatformResult = {
+    endpoint: defaults.endpoint,
+  };
+  return Promise.resolve(platformConfig);
+}
+
+/**
+ * Get all state="ACTIVE" and type="CODE" repositories from gerrit
+ */
+export async function getRepos(): Promise<string[]> {
+  logger.debug(`getRepos()`);
+  return await client.getRepos();
+}
+
+/**
+ * Clone repository to local directory
+ * @param config
+ */
+export async function initRepo({
+  repository,
+  endpoint,
```

Comment:
```suggestion
```

i don't think endpoint can have changed here. it's only happen on github for renovate pro, but will be removed soon too.